### PR TITLE
feat(vue): support keepalive

### PR DESCRIPTION
**Pull request by LittleSound**
When using `useHead` in components that use `<KeepAlive >`, `useHead` works even after the page is disabled (`onDeactivated`). The purpose of this PR is to adapt the lifecycle function related to Vue KeepAlive.

### CODE REVIEW

**Comment by LittleSound** [comment]
I am trying to add unit tests for this part of the code so that I can better indicate the intent of this PR and make sure it works correctly. It will take me some time to work out how to write it.

File: test/vue/util.ts [hunk 1]
```diff
@@ -33,3 +35,41 @@ export async function ssrRenderOptionsHead(input: any) {
...
+
+export function mount<V extends Component>(Comp: V) {
+  const el = document.createElement('div')
+  const app = createApp(Comp)
+
+  const unmount = () => app.unmount()
+  const comp = app.mount(el) as any as VM<V>
+  comp.unmount = unmount
+  return comp
+}
```
Comment (LittleSound) [inline test/vue/util.ts:50]: This unit test has to be configured with `environment: 'jsdom'` to work, but this causes problems with some other unit tests, and I need to look into how to turn on the `environment: 'jsdom'` option for this file alone or some other option.

**Comment by LittleSound** [comment]
I added unit tests for this PR

**Comment by harlan-zw** [comment]
Thank you @LittleSound sound, this is a great fix, nice tests too.

I'll merge now, will do some testing and release likely tomorrow.

**Comment by harlan-zw** [comment]
Merged into 1.0.21. This seemed more like a bug fix than a new feature

**Comment by LittleSound** [comment]
Thank you, support keepalive solved some bugs

**Pull request merged**